engine/src/conversion/analysis/struct_accessors.rs

```diff
@@ -0,0 +1,142 @@
+                            this: *const #struct_type
+                        };
+
+                        // TODO: convert arrays to pointers?
+                        let field_type = &field.ty;
+
+                        accessors.push(Api::Function {
+                            name: ApiName::new_from_qualified_name(accessor_name),
+                            fun: Box::new(FuncToConvert {
+                                provenance: Provenance::SynthesizedOther,
+                                ident,
+                                doc_attrs: Vec::new(),
+                                inputs: [fnarg].into_iter().collect(),
+                                variadic: false,
+                                output: parse_quote! {
+                                    -> #field_type
+                                },
+                                vis: parse_quote! { pub },
+                                virtualness: crate::conversion::api::Virtualness::None,
+                                cpp_vis: crate::conversion::api::CppVisibility::Public,
+                                special_member: Some(SpecialMemberKind::GeneratedAccessor),
+                                unused_template_param: false,
+                                references: Default::default(),
+                                original_name: None,
+                                self_ty: None,
+                                synthesized_this_type: None,
+                                add_to_trait: None,
+                                synthetic_cpp: Some((body, CppFunctionKind::Function)),
+                                is_deleted: false,
+                            }),
+                            analysis: (),
+                        })
+                    }
+                }
+                // TODO: is it even possible for these to occur?
+                syn::Fields::Unnamed(_) => todo!(),
+                syn::Fields::Unit => todo!(),
+            };
+
+            // Generate the accessors (if any) + the struct itself
+            Ok(Box::new(accessors.into_iter().chain(std::iter::once(
+                Api::Struct {
+                    name: struct_name,
+                    details,
+                    analysis,
+                },
+            ))))
+        },
+        Api::enum_unchanged,
+        Api::typedef_unchanged,
+    );
+
+    results
+}
+
+fn should_generate_accessor(field: &Field) -> bool {
+    // Don't generate accessors for non-public fields
+    if !matches!(field.vis, Visibility::Public(_)) {
+        return false;
+    }
+
+    // Don't generate accessors for cpp reference fields
+    // TODO: is this restriction necessary?
+    if field.attrs.iter().any(is_cpp_reference) {
+        return false;
+    }
+
+    // Don't generate accessors for "fake" bindgen fields which wouldn't appear directly in the C++ struct
+    // TODO: is there a better way to tell which fields these are?
```

Comment:
I don't know of a better way right now. In principle I like the idea of bindgen-specific knowledge being isolated into the `parse` bit of the codebase and I am gradually heading in that direction, but there's no consistency there yet. This is fine.